### ISSUE TITLE
Show/hide demo broken

### ISSUE THREAD

**Issue by abbottmw** [VinceG/twitter-bootstrap-wizard#209]
On the demo where you can show/hide tabs or steps, after you hide a tab, the next and previous do not work correctly.  It skips over steps.

http://vinceg.github.io/twitter-bootstrap-wizard/examples/remove-step.html#

**Comment by VinceG** [comment]
@abbottmw Yeah, i think because it does not call the refresh to reset the index count.

**Comment by abbottmw** [comment]
The :visible fix I added *should* resolve this.

**Comment by Adam-78** [comment]
The show/hide methods do not work even with the  :visible fix added?  Doing the following should show tab 3 and its tab panel - nothing happens?

    $('#rootwizard').bootstrapWizard('display', 3);

Currently I'm having to revert to the standard bootstrap method:

      $('#rootwizard li:eq(2) a').tab('show') // Select third tab (0-indexed)

When can we expect a fix?

**Comment by VinceG** [comment]
@Adam-78 

Try

`$('#rootwizard').bootstrapWizard('show', 2);`

**Comment by Adam-78** [comment]
Many thanks. That worked like a charm.


________________________________

**Comment by Adam-78** [comment]
Is it possible to call the show method from inside the onInit event. The following doesnt work:

     $('#rootwizard').bootstrapWizard({
           'onInit': function (tab, navigation, index) {
                var lastTabIndex = $("#LastTabIndex").val();
                if (lastTabIndex)
                      $('#rootwizard').bootstrapWizard('show', lastTabIndex);

            }
       });

**Comment by VinceG** [comment]
@Adam-78 without the condition does `$('#rootwizard').bootstrapWizard('show', x);` work at all inside onInit?

**Comment by Adam-78** [comment]
I get console error:
 
    TypeError: this.data(...) is undefined

In the debugger it highlights the following line:

      return this.data('bootstrapWizard')[options](args);

I think it's because I'm trying to call the following from within `$('#rootwizard').bootstrapWizard({...})`

    $('#rootwizard').bootstrapWizard('show', x);

**Comment by VinceG** [comment]
@Adam-78 yeah it's impossible since onInit is called right after it's initialized.

i was able to make it work with the following

```
$('#rootwizard').bootstrapWizard({
        'onInit': function (tab, navigation, index) {
            setTimeout(function() {
              $('#rootwizard').bootstrapWizard('show', 2);
            }, 1);
        }
      });
```
